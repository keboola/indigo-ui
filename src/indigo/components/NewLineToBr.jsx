import React from 'react';
import PropTypes from 'prop-types';

class NewLineToBr extends React.Component {
  render() {
    const lines = this.props.text.split('\n');
    let result = [];

    lines.forEach((value, index) => {
      result.push(<span key={index}>{value}</span>);
      if (index < lines.length - 1) {
        result.push(<br key={index + '-br'} />);
      }
    });

    return (
      <span>{result}</span>
    );
  }
}

NewLineToBr.propTypes = {
  text: PropTypes.string.isRequired
};

export default NewLineToBr;
