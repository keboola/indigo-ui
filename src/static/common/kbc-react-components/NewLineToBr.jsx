import React, {PropTypes, DOM} from 'react';
import {List} from 'immutable';

export default React.createClass({
  propTypes: {
    text: PropTypes.string.isRequired
  },

  render() {
    return (
      <span>{this.lines()}</span>
    );
  },

  lines() {
    return List(this.props.text.split("\n"))
      .map((value, index) => {
        return (
          <span key={index}>{value}</span>
        );
      })
      .interpose(DOM.br());
  }

});
