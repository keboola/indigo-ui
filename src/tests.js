import renderer from 'react-test-renderer';

export const snapshot = (component) => {
  const tree = renderer
    .create(component)
    .toJSON();
  expect(tree).toMatchSnapshot();
};
