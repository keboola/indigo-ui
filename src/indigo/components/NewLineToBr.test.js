import React from 'react';

import { snapshot } from "../../tests";
import NewLineToBr from './NewLineToBr';

describe('<NewLineToBr />', () => {

  it('NewLineToBr - default', () => {
    snapshot(<NewLineToBr text={"one\ntwo\nthree"}/>);
  });

  it('NewLineToBr - no newline', () => {
    snapshot(<NewLineToBr text={"one"}/>);
  });

});
