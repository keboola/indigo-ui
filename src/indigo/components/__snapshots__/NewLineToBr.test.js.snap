// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewLineToBr /> NewLineToBr - default 1`] = `
<span>
  <span>
    one
  </span>
  <br />
  <span>
    two
  </span>
  <br />
  <span>
    three
  </span>
</span>
`;

exports[`<NewLineToBr /> NewLineToBr - no newline 1`] = `
<span>
  <span>
    one
  </span>
</span>
`;
